prov/efa/src/efa_user_info.c

```diff
@@ -326,24 +326,41 @@ int efa_user_info_alter_rxr(struct fi_info *info, const struct fi_info *hints)
 		 * which means FI_MR_HMEM implies FI_MR_LOCAL for device buffer.
 		 */
 		if (hints->caps & FI_HMEM) {
-			if (ofi_hmem_p2p_disabled()) {
-				EFA_WARN(FI_LOG_CORE,
-					"FI_HMEM capability currently requires peer to peer support, which is disabled.\n");
-				return -FI_ENODATA;
-			}
-			/* TODO: remove the rdma checks once FI_HMEM w/o p2p is supported */
-
-			if (!efa_device_support_rdma_read()) {
-				EFA_WARN(FI_LOG_CORE,
-				        "FI_HMEM capability requires RDMA, which this device does not support.\n");
-				return -FI_ENODATA;
+			/*
+			 * FI_HMEM_CUDA does not require p2p if the user explicitly enables
+			 * CUDA memory transfers via FI_HMEM_CUDA_ENABLE_XFER
+			 */
+			if (hmem_ops[FI_HMEM_CUDA].initialized &&
+					cuda_get_xfer_setting() == CUDA_XFER_ENABLED) {
```

Comment:
indentation seems off.

Reply:
ack, reduced the extra indentations in this file